astropy/units/quantity.py

```diff
@@ -275,6 +275,14 @@ def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
     def __array_finalize__(self, obj):
         self._unit = getattr(obj, '_unit', None)
 
+        # Copy info (except for a scalar output for performance reasons) if the
+        # original had `info` defined.  Because of the way the InfoDescriptor
+        # works, `'info' in obj.__dict__` is False until the `info` attribute
+        # is accessed or set.  Note that `obj` can be an ndarray which doesn't
+        # have a `__dict__`.
+        if self.ndim > 0 and hasattr(obj, '__dict__') and 'info' in obj.__dict__:
```

Comment:
No, not yet, but will do. I do know that `hasattr` and `getattr(.., ..., default)` effectively use `try/except` internally, so that the case for not having the attribute is always rather slow.